docs/hdf5-file-specification.ipynb

```diff
@@ -0,0 +1,698 @@
+    "\n",
+    "#### The second group is called `mesh`. It contains:\n",
+    "\n",
+    "\n",
+    "**attributes**\n",
+    "\n",
+    "- `bc` (`str`): information about boundary conditions as defined in `discretisedfield.Mesh.bc`\n",
+    "- `n` (`list[int], length region.ndim`):  number of cells along the different spatial directions\n",
+    "\n",
+    "**datasets** (optional, only present if subregions are defined)\n",
+    "\n",
+    "- `subregion_names` (`list[str]`): names of the individual subregions defined for the field,  only present if subregions are defined\n",
+    "- `subregions` (`array[<number of subregions>, region.ndim*2]`): coordinates of `discretisedfield.Region.pmin` and `discretisedfield.Region.pmax`, subregions and subregion names are combined based on their order, only present if subregions are defined |\n",
+    "\n",
+    "**groups**\n",
+    "\n",
+    "- `region` (`h5py.Group`): region on which the mesh is defined\n",
+    "\n",
+    "#### The third group is called `region`. It contains:\n",
+    "\n",
+    "**attributes**\n",
+    "\n",
+    "- `dims` (`list[str], length ndim`): names of spatial dimensions\n",
+    "- `ndim` (`int`): number of spatial dimensions\n",
+    "- `pmin` (`list[number], length ndim`): coordinates of the minimum point `discretisedfield.Region.pmin`\n",
+    "- `pmax` (`list[number], length ndim`): coordinates of the minimum point `discretisedfield.Region.pmin`\n",
+    "- `tolerance_factor` (`float`): `discretisedfield.Region.tolerance_factor`\n",
+    "- `units` (`list[str], length ndim`): units of the spatial dimensions\n",
+    "\n",
+    "#### Examples:\n",
+    "**1d field with 1 vdim**"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "id": "cf5faefe-a6ce-44f5-bd21-5854d6c0f4a1",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [
+    {
+     "data": {
+      "text/html": [
+       "<strong>Field</strong>\n",
+       "<ul>\n",
+       "  \n",
+       "  <li><strong>Mesh</strong>\n",
+       "<ul>\n",
+       "  <li><strong>Region</strong>\n",
+       "<ul>\n",
+       "  <li>pmin = [0]</li>\n",
+       "  <li>pmax = [20]</li>\n",
+       "  <li>dims = ['x']</li>\n",
+       "  <li>units = ['m']</li>\n",
+       "</ul></li>\n",
+       "  <li>n = [20]</li>\n",
+       "  </ul></li>\n",
+       "  <li>nvdim = 1</li>\n",
+       "  </ul>"
+      ],
+      "text/plain": [
+       "Field(Mesh(Region(pmin=[0], pmax=[20], dims=['x'], units=['m']), n=[20]), nvdim=1)"
+      ]
+     },
+     "execution_count": 4,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "mesh = df.Mesh(p1=0, p2=20, n=20)\n",
+    "field_1d_scalar = df.Field(mesh, nvdim=1, value=0)\n",
+    "field_1d_scalar"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "id": "93791e84-c69a-424b-9110-aa4f68587fc1",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import numpy as np"
```

Comment:
Why do we import numpy again here?

Reply:
Seems to be a left-over from some checks. Removed now.